site/src/request_handlers/graph.rs

```diff
@@ -140,9 +145,13 @@ async fn create_graphs(
     }))
 }
 
-fn artifact_ids_for_range(ctxt: &SiteCtxt, start: Bound, end: Bound) -> Vec<ArtifactId> {
+fn master_artifact_ids_for_range(ctxt: &SiteCtxt, start: Bound, end: Bound) -> Vec<ArtifactId> {
     let range = ctxt.data_range(start..=end);
-    range.into_iter().map(|c| c.into()).collect()
+    range
+        .into_iter()
+        .filter(|c| matches!(c.r#type, CommitType::Master))
```

Comment:
We probably want to explicitly include the endpoints (start/end) if they're commit SHAs, even if they are try commits. The comparison page generates links like https://perf.rust-lang.org/index.html?start=2022-06-12&end=b3f4c3119957aa0a250cab08ab586b7a9a680ef1&benchmark=ctfe-stress-5&profile=opt&scenario=full&stat=instructions:u

for which the end bound is the end commit on the compare page, and that can (often) be a try commit. It's not a big deal to drop it, but it seems annoying and probably not what we want.

Reply:
Good catch, thanks. Hopefully it is fixed now.